Inductor cpp wrapper: support sympy.Expr as input

chunyuan-w opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #101257

Leverage the logic in https://github.com/pytorch/pytorch/pull/95533 to get the `dtype` of `sympy.Expr` and support it as graph input in the cpp wrapper.


cc @soumith @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

chunyuan-w committed:
Inductor cpp wrapper: support sympy.Expr as input

[ghstack-poisoned]

chunyuan-w committed:
Update on "Inductor cpp wrapper: support sympy.Expr as input"




cc soumith voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

chunyuan-w committed:
Update on "Inductor cpp wrapper: support sympy.Expr as input"




cc soumith voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

chunyuan-w committed:
Update on "Inductor cpp wrapper: support sympy.Expr as input"


Leverage the logic in https://github.com/pytorch/pytorch/pull/95533 to get the `dtype` of `sympy.Expr` and extract the logic into a utility function.


cc soumith voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

chunyuan-w committed:
Update on "Inductor cpp wrapper: support sympy.Expr as input"


Leverage the logic in https://github.com/pytorch/pytorch/pull/95533 to get the `dtype` of `sympy.Expr` and support it as graph input in the cpp wrapper.


cc soumith voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

Skylion007 commented on torch/_inductor/utils.py:
```diff
@@ -894,6 +894,16 @@ def is_cpu_device(inputs):
     )
 
 
+def get_sympy_Expr_dtype(val):
```
```suggestion
def get_sympy_Expr_dtype(val : sympy.Expr) -> torch.dtype:
```
No reason not to add type hints here, right?

chunyuan-w replied:
Thanks for the suggestion. Type hints are added.

chunyuan-w committed:
Update on "Inductor cpp wrapper: support sympy.Expr as input"


Leverage the logic in https://github.com/pytorch/pytorch/pull/95533 to get the `dtype` of `sympy.Expr` and support it as graph input in the cpp wrapper.


cc soumith voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

chunyuan-w committed:
Update on "Inductor cpp wrapper: support sympy.Expr as input"


Leverage the logic in https://github.com/pytorch/pytorch/pull/95533 to get the `dtype` of `sympy.Expr` and support it as graph input in the cpp wrapper.


cc soumith voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

chunyuan-w commented:
@pytorchbot merge
